doc/md_docs.go

```diff
@@ -50,8 +66,8 @@ func GenMarkdown(cmd *cobra.Command, w io.Writer) error {
 	return GenMarkdownCustom(cmd, w, func(s string) string { return s })
 }
 
-// GenMarkdownCustom creates custom markdown output.
-func GenMarkdownCustom(cmd *cobra.Command, w io.Writer, linkHandler func(string) string) error {
```

Comment:
I really like the idea. But this would break API compatibility. You may create a new function `GenMarkdownWithOpts` and call it from here. Please note this function as being deprecated in the godoc. Please make sure to also supply to end users (and use ourselves) default versions of the GenMarkdownOptions and GenMarkdownTreeOptions. So that an end user can easily use the new functions and get the default values.

Reply:
All good points. I’ll be happy to do so. :)